### ISSUE TITLE
Clarify media-source frame rates (frames and framesPerSecond)

### ISSUE THREAD

**Issue by henbos** [w3c/webrtc-stats#708]
RTCVideoSourceStats contains [frame counters](https://w3c.github.io/webrtc-stats/#videosourcestats-dict*) as "frames originating from this source".

But the spec is not clear about which point in the pipeline this is measured. If you measure this as early to camera capture as possible you may see more frames than if you measure this at the entry to WebRTC in rare cases such as running out of frame buffers or other errors causing us to drop frames.

We should clarify the spec to say that...

1. These are the frames that have reached WebRTC. E.g. the delta between this input fps and the encoder fps would be frames that have been dropped by the webrtc encoder, not frames that were dropped between capture process and renderer process etc.

This matches what libwertc is already doing.

2. Clarify that these are the frames prior to adaptation. Adaptation is an encoder implementation detail, so frames dropped due to adaptation should be part of the delta between input fps and encoder fps.

libwebrtc has a [known bug](https://bugs.chromium.org/p/webrtc/issues/detail?id=12501) where we count the frames after adaptation, but this is wrong since it exposes an implementation detail that does not make sense on paper

**Comment by henbos** [comment]
The idea based on different spec discussions (e.g. [this](https://github.com/w3c/mediacapture-extensions/pull/77)) is that we should have a counter for camera fps on the track and then each frame consuming use case should have its own frame counter, for example media-source.frames being a measurement that happens later than track.framesCaptured()

**Comment by henbos** [comment]
Editorial label because I'm not proposing to change what was, as far as I understand, always the intent of the spec. E.g. we always counted this in webrtc, and I always considered post adaptation a bug.

**Comment by vr000m** [comment]
This makes sense, agree it is editorial. And this is kinda expected, as we start to have more frames* metrics along the send and receive pipeline, we will notice that the definitions become more nuanced.

**Comment by henbos** [comment]
Here's another edge case: Chrome has a feature where, to save on performance and bandwidth, screenshare tracks stop capturing new frames when the content is static. But to ensure good QP values and frames not getting lost, the enocoder will repeat frames.

My opinion is that any frame that gets input to the encoder is considered media-source frames, even if it includes repeated frames.

**Comment by henbos** [comment]
Now that we have [MediaStreamTrack Statistics](https://w3c.github.io/mediacapture-extensions/#the-mediastreamtrackvideostats-interface), i.e. `track.stats`, we do have stats from capture to delivery to sinks.

So I think we should re-think this part:

> Clarify that these are the frames prior to adaptation. Adaptation is an encoder implementation detail, so frames dropped due to adaptation should be part of the delta between input fps and encoder fps.

If the frames are prior to adaptation, they're almost the same think as `track.stats.deliveredFrames`. Alternatively, if we update the spec to match what the [implementation is already doing](https://crbug.com/webrtc/12501), i.e. they are frame counters _after adaptation_, then we have a good measure on what WebRTC is trying to encode with.

Either option is valid, but since we already have `track.stats`, it might actually make sense to say that `media-source` is AFTER adaptation. This gives the app more information.

**Comment by fippo** [comment]
👍 this makes it more "peerconnection-media-source" but changing names is hard.

**Comment by henbos** [comment]
Maybe reword from "input to the sender" to "input to the encoder" since the former sounds like track stats
